app/src/main/java/fr/free/nrw/commons/utils/PermissionUtils.java

```diff
@@ -0,0 +1,19 @@
+package fr.free.nrw.commons.utils;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.provider.Settings;
+
+public class PermissionUtils {
+
+    //Can be used by any activity which requires a permission which has been blocked(marked never ask again by the user)
```

Comment:
Very helpful comment! :) But could you please write this in Javadoc format to take advantage of IDE support?

Reply:
Should have been there, will do it